gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcOperationApi.java

```diff
@@ -0,0 +1,108 @@
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.grpc;
+
+import com.google.api.core.ApiFunction;
+import com.google.api.gax.rpc.OperationApi;
+import com.google.api.gax.rpc.OperationSnapshot;
+import com.google.api.gax.rpc.TranslatingUnaryCallable;
+import com.google.api.gax.rpc.UnaryCallable;
+import com.google.longrunning.CancelOperationRequest;
+import com.google.longrunning.DeleteOperationRequest;
+import com.google.longrunning.GetOperationRequest;
+import com.google.longrunning.Operation;
+import com.google.longrunning.stub.OperationsStub;
+import com.google.protobuf.Empty;
+
+public class GrpcOperationApi implements OperationApi {
+  private final OperationsStub operationsStub;
+
+  public GrpcOperationApi(OperationsStub operationsStub) {
+    this.operationsStub = operationsStub;
+  }
+
+  @Override
+  public UnaryCallable<String, OperationSnapshot> getOperationCallable() {
+    return TranslatingUnaryCallable.create(
+        operationsStub.getOperationCallable(),
+        new ApiFunction<String, GetOperationRequest>() {
+          @Override
+          public GetOperationRequest apply(String request) {
+            return GetOperationRequest.newBuilder().setName(request).build();
+          }
+        },
+        new ApiFunction<Operation, OperationSnapshot>() {
+          @Override
+          public OperationSnapshot apply(Operation operation) {
+            return GrpcOperationSnapshot.create(operation);
+          }
+        });
+  }
+
+  @Override
+  public UnaryCallable<String, Void> cancelOperationCallable() {
+    return TranslatingUnaryCallable.create(
+        operationsStub.cancelOperationCallable(),
+        new ApiFunction<String, CancelOperationRequest>() {
+          @Override
+          public CancelOperationRequest apply(String request) {
+            return CancelOperationRequest.newBuilder().setName(request).build();
+          }
+        },
+        new ApiFunction<Empty, Void>() {
```

Comment:
It seems like a no-op transformer. Is it possible inside `TranslatingUnaryCallable` to just not do any transformation in such cases (this allows to avoid doing heavy ApiFutures.transform() when it is not required).

Reply:
Unfortunately you can't just return an `Empty` as a `Void` (the first will have a default instance, the second has to be `null`).